pkg/stream/redis.go

```diff
@@ -0,0 +1,74 @@
+package stream
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/blocknative/dreamboat/pkg/structs"
+	"github.com/go-redis/redis/v8"
+)
+
+type RedisPubsub struct {
+	Redis *redis.Client
+}
+
+func (r *RedisPubsub) Publish(ctx context.Context, topic string, data []byte) error {
+	return r.Redis.Publish(ctx, topic, data).Err()
+}
+
+func (r *RedisPubsub) Subscribe(ctx context.Context, topic string) (chan []byte, error) {
+	pubsub := r.Redis.Subscribe(ctx, topic)
+
+	sub := make(chan []byte)
```

Comment:
Consider making this a `chan string` instead of `chan byte`, and doing the `[]byte` conversion at the call site (in `RunSubscriber`).  There's a good chance this will avoid escaping the stack.